File: app/src/main/java/soup/movie/ui/main/plan/PlanViewState.java
package soup.movie.ui.main.plan;

import android.support.annotation.NonNull;

import java.util.List;

import soup.movie.data.model.Movie;
import soup.movie.ui.BaseViewState;

interface PlanViewState extends BaseViewState {

    class LoadingState implements PlanViewState {

        LoadingState() {
        }

        @Override
        public String toString() {
            return "LoadingState{}";
        }
    }

    class DoneState implements PlanViewState {

        private final List<Movie> movies;

        DoneState(@NonNull List<Movie> movies) {
            this.movies = movies;
        }

        @NonNull
        public List<Movie> getMovies() {
            return movies;
        }

        @Override
        public String toString() {
            return "DoneState{" +
                    "movies=" + movies +
                    '}';
        }
    }
}

File: app/src/main/java/soup/movie/ui/main/plan/PlanViewState.kt
package soup.movie.ui.main.plan

import androidx.annotation.Keep
import soup.movie.data.model.Movie

sealed class PlanViewState {

    @Keep
    object LoadingState : PlanViewState() {

        override fun toString(): String = javaClass.simpleName
    }

    @Keep
    object ErrorState : PlanViewState() {

        override fun toString(): String = javaClass.simpleName
    }

    @Keep
    data class DoneState(val movies: List<Movie>) : PlanViewState()
}
